services/src/main/java/org/keycloak/authentication/authenticators/browser/CookieAuthenticator.java

```diff
@@ -83,6 +84,15 @@ public void close() {
 
     }
 
+    protected boolean requireReauthentication(UserSessionModel userSession, ClientSessionModel clientSession) {
+        return isPromptLogin(clientSession) || isAuthTimeExpired(userSession, clientSession);
+    }
+
+    protected boolean isPromptLogin(ClientSessionModel clientSession) {
```

Comment:
I really don't like this approach - it's hard coding OIDC specifics into the cookie authenticator. We need a generic approach in the authenticator framework instead.